### ISSUE TITLE
ignore "Avoid restricted flags" not respected in test/Spec.hs

### ISSUE THREAD

**Issue by JasonMFry** [ndmitchell/hlint#1320]
I suspect this is user error on my part, but I have tried everything I can think of and I'm still getting an hlint error when I don't want to.

```text
test/Spec.hs|1| 1-68: Warning: Avoid restricted flags
|| Found:
||   {-# OPTIONS_GHC -F -pgmF hspec-discover -optF --module-name=Spec #-}
|| Note: may break the code, may break the code, may break the code, may break the code, may break the code
|| 
|| 1 hint
|| make: *** [hlint] Error 1
```

```haskell
-- test/Spec.hs
{-# OPTIONS_GHC -F -pgmF hspec-discover -optF --module-name=Spec #-}
```

```yaml
# .hlint.yaml
- ignore:
    name:
      - Avoid restricted flags
      - Avoid restricted function
      - Avoid restricted module
      - Avoid restricted qualification
    within:
      - "**.*Spec"
      - Spec # I have tried many things here, including Spec.hs, test/Spec.hs, test.Spec.hs, etc etc
```

**Comment by zliu41** [comment]
@JasonMFry  I _think_ "within" matches module names, not file names.

**Comment by JasonMFry** [comment]
> @JasonMFry I _think_ "within" matches module names, not file names.

@zliu41 you're right, but I expected "**.*Spec" to match "test/Spec.hs" and when it didn't I tried everything. So this indicates that hlint is getting the module name from within the module itself, rather than looking at the file name and constructing the module name from that. So is there any way to ignore this rule inside test/Spec.hs?

**Comment by zliu41** [comment]
Yeah, so instead of using `.hlint.yaml`, you can put an `ANN` inside `test/Spec.hs`. See README.md for some examples.

I do think being able to match on file names is useful though, e.g., it would make it easier to turn off a hint within a dir.

**Comment by JasonMFry** [comment]
@zliu41 thanks for the input, I was hoping this was possible but using `{-# ANN module "HLint: ignore Avoid restricted flags" #-}` does work of course.

**Comment by ndmitchell** [comment]
I would have expected `**.Spec` to match `Spec` - does that work? Does just `Spec` alone there work? HLint does indeed use the module name as reported by the module, rather than synthesising one including directory components.
